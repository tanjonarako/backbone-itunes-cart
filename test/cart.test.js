import Backbone from "backbone";
import CartView from "../src/views/cart";
import CartCollection from "../src/models/cart";

describe('views:CartView', function() {
    const list = new CartCollection([
        {trackId: '1', trackName: 'Shape of you', artistName: 'Ed Sheeran'},
        {trackId: '2', trackName: 'Perfect', artistName: 'Ed Sheeran'}
    ]);

    it('should render a list', function() {
        const cartView = new CartView({collection: list}).render();

        expect(cartView.$el.html()).toMatchSnapshot();
    });

    it('should call add to cart', function() {
        const spy = jest.fn();
        const cartView = new CartView({collection: list, handleRemoveFromCart: spy}).render();

        cartView.$el.find('.js-btn-remove[data-id=1]').click();

        expect(spy).toBeCalledWith('1');
    });
});
