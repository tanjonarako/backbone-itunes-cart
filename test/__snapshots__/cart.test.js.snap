// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views:CartView should render a list 1`] = `
"
    <h2>Cart</h2>
    <table class=\\"table table-bordered\\">
        <thead class=\\"thead-light\\">
            <tr>
                <th scope=\\"col\\">Track name</th>
                <th scope=\\"col\\">Remove to cart</th>
            </tr>
        </thead>
        <tbody>
            
            <tr>
                <td>Shape of you</td>
                <td><button type=\\"button\\" data-id=\\"1\\" class=\\"btn btn-secondary btn-remove cursor-pointer\\">Remove</button></td>
            </tr>
            
            <tr>
                <td>Perfect</td>
                <td><button type=\\"button\\" data-id=\\"2\\" class=\\"btn btn-secondary btn-remove cursor-pointer\\">Remove</button></td>
            </tr>
            
        </tbody>
    </table>

"
`;
